pkg/handlers/update.go

```diff
@@ -437,3 +370,145 @@ func (h *Handler) GetAdminConsoleUpdateStatus(w http.ResponseWriter, r *http.Req
+		return
+	}
+
+	// Check content type to determine if this is an airgap request
+	contentType := strings.Split(r.Header.Get("Content-Type"), ";")[0]
+	contentType = strings.TrimSpace(contentType)
+
+	if contentType == "multipart/form-data" {
+		// Use the shared airgap processing function
+		handleAirgapUpdate(w, r, a, skipPreflights, false, false)
+		return
+	}
+
+	update := upstreamtypes.Update{
+		ChannelID:   channelID,
+		Cursor:      channelSequenceStr,
+		AppSequence: nil, // nil creates a new version
+	}
+
+	// Download release from replicated app
+	finalSequence, err := upstream.DownloadUpdate(a.ID, update, skipPreflights, false)
+	if err != nil {
+		cause := errors.Cause(err)
+		if _, ok := cause.(util.ActionableError); ok {
+			upstreamUpdateResponse.Error = cause.Error()
+		} else {
+			upstreamUpdateResponse.Error = "failed to download upstream update"
+		}
+		logger.Error(errors.Wrap(err, "failed to download upstream update"))
+		JSON(w, http.StatusInternalServerError, upstreamUpdateResponse)
+		return
+	}
+
+	if finalSequence != nil {
+		logger.Infof("Successfully downloaded upstream update for app %s, new sequence: %d", appSlug, *finalSequence)
+	} else {
+		logger.Infof("Successfully downloaded upstream update for app %s", appSlug)
+	}
+
+	upstreamUpdateResponse.Success = true
+	JSON(w, http.StatusOK, upstreamUpdateResponse)
+}
+
+// handleAirgapUpdate processes the entire airgap update flow
+func handleAirgapUpdate(w http.ResponseWriter, r *http.Request, app *apptypes.App, skipPreflights bool, deploy bool, skipCompatibilityCheck bool) {
+	if !app.IsAirgap {
+		logger.Error(errors.New("not an airgap app"))
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte("Cannot update an online install using an airgap bundle"))
+		return
+	}
+
+	rootDir, err := os.MkdirTemp("", "kotsadm-airgap")
+	if err != nil {
+		logger.Error(errors.Wrap(err, "failed to create temp dir"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	defer os.RemoveAll(rootDir)
+
+	formReader, err := r.MultipartReader()
+	if err != nil {
+		logger.Error(errors.Wrap(err, "failed to get multipart reader"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	for {
+		part, err := formReader.NextPart()
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			logger.Error(errors.Wrap(err, "failed to get next part"))
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+
+		fileName := filepath.Join(rootDir, part.FormName())
+		file, err := os.Create(fileName)
+		if err != nil {
+			logger.Error(errors.Wrap(err, "failed to create file"))
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		defer file.Close()
+
+		_, err = io.Copy(file, part)
+		if err != nil {
+			logger.Error(errors.Wrap(err, "failed to copy part data"))
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		file.Close()
+	}
+
+	finishedChan := make(chan error)
```

Comment:
this can cause a deadlock

```suggestion
	finishedChan := make(chan error, 1)
```